UNRELEASED.md

```diff
@@ -24,4 +24,6 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Code quality
 
+- Updated how we use React.Component and React.PureComponent to help treeshakability ([#3133](https://github.com/Shopify/polaris-react/pull/3133))
```

Comment:
There's a bit more that's covered in this PR, I think this item would benefit from summarizing the other changes as well.